lib/srv/alpnproxy/common/kube.go

```diff
@@ -37,6 +39,16 @@ func TeleportClusterFromKubeLocalProxySNI(serverName string) string {
 	return teleportCluster
 }
 
+// KubeClusterFromKubeLocalProxySNI returns Kubernetes cluster name from SNI.
+func KubeClusterFromKubeLocalProxySNI(serverName string) (string, error) {
+	kubeCluster, _, _ := strings.Cut(serverName, ".")
```

Comment:
should we care about the teleport cluster?
It's possible you have the same cluster name in two different teleport clusters

Reply:
[I read the teleport cluster](https://github.com/gravitational/teleport/blob/341eac2c87002135b93c340b0324882a476d9846/lib/srv/alpnproxy/kube.go#L256) using `TeleportClusterFromKubeLocalProxySNI()` before reading the kube cluster. 
Then `m.certReissuer()` is called with that pair of teleport + kube cluster. 

Do you mean something else?